### DISCUSSION TITLE
Sort a List of objects based on a Method (c#)

### DISCUSSION THREAD

**Discussion by benasci** [community/community#21518] [Other Feature Feedback, Questions, & Ideas]
<p>I’m quite new to coding and am trying to write code for an API for Revit in c#.</p>
<p>I have a List of <em>Objects</em> :  List&lt;Element&gt;</p>
<p>and a List of <em>double</em> : Dist&lt;double&gt;</p>
<p>Each double in Dist corresponds to an object in List in the same order.</p>
<p>I need to sort the List based on doubles from Dist.</p>
<p>this is an Api and I can’t simply define a new property for object and sort by that property.</p>
<p>How can i do this? Is this possible with Linq, Query ?</p>
<p>Thanks</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>It’s not clear from your question if you need to sort in-place, or can return a new list of the sorted Elements.</p>
<p>I can think of several ways to do this, but they all require building a new data set and sorting that, to build a resulting sorted list of Element.</p>
<p>(the code below is pseudo-C#, I’m doing this off-the-cuff)</p>
<p>For instance, you would build a List&lt;Tuple&lt;double, int&gt;&gt; which is your second list (Dist?) and the index within the list, and then sort it: </p>
<pre><code>var sortlist = new List&lt;Tuple&lt;double,int&gt;&gt;();
for (int dex = 0; dex &lt; Dist.Length; dex++)
    sortlist.add(new Tuple&lt;double,int&gt;(Dist[i], i);
sortlist.Sort((x, y) =&gt; y.Item1.CompareTo(x.Item1));
</code></pre>
<p>Now sortlist is sorted by your double values (Item1 in the tuple). You can then build a sorted list of Element because the sortlist tuple has the matching index.</p>
<pre><code>var sortedElements = new List&lt;object&gt;();
for (int dex = 0; dex &lt; sortlist.Length; dex++)
    sortedElements.add( originalElements[sortlist[dex].Item2 );
</code></pre>
<p>sortedElements is now a List&lt;Element&gt; which is sorted in Dist order. originalElements being your original List&lt;Element&gt;; Item2 in the tuple being the original index.</p>
<p>Maybe someone smarter than I can suggest a better approach.</p>

**Comment by benasci** [comment]
<p>Thanks</p>
<p>i got the idea. Now i’ll try to write the code</p>
